Sources/BraveWallet/WalletUserAssetManager.swift

```diff
@@ -100,16 +102,25 @@ public class WalletUserAssetManager: WalletUserAssetManagerType {
     return allUserSpamAssets.sorted(by: { $0.sortOrder < $1.sortOrder })
   }
   
+  public func getAllUserDeletedNFTs() -> [WalletUserAsset] {
+    WalletUserAsset.getAllUserDeletedUserAssets() ?? []
+  }
+  
   public func getUserAsset(_ asset: BraveWallet.BlockchainToken) -> WalletUserAsset? {
     WalletUserAsset.getUserAsset(asset: asset)
   }
   
   public func addUserAsset(_ asset: BraveWallet.BlockchainToken, completion: (() -> Void)?) {
-    guard WalletUserAsset.getUserAsset(asset: asset) == nil else {
-      completion?()
-      return
+    if let existedAsset = WalletUserAsset.getUserAsset(asset: asset) {
+      if existedAsset.isDeletedByUser { // this asset was added before but user marked as deleted after
+        WalletUserAsset.updateUserAsset(for: asset, visible: true, isSpam: false, isDeletedByUser: false, completion: completion)
+      } else { // this asset exists, either in `Collected` or `Hidden` based on its `visible` value
```

Comment:
nit: I don't think we want to reference UI in comment here; could be hard to follow later if UI is updated and comment is not.
```suggestion
      } else { // this asset already exists
```